inginious/frontend/pages/course_user_settings.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+"""
+Module definition for CourseUserSettingPage class
+
+This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+more information about the licensing of this file.
+"""
+
+import flask
+from inginious.common.field_types import FieldTypes
+from inginious.frontend.pages.utils import INGIniousAuthPage
+
+
+class CourseUserSettingPage(INGIniousAuthPage):
+    """
+    Class definition for CourseUserSettingPage
+    """
+
+    def GET_AUTH(self, courseid):
+        """ GET request """
+        username = self.user_manager.session_username()
+        if not self._is_accessible(courseid, username):
+            return self.template_helper.render("course_unavailable.html")
+        current_user = self.database.users.find_one(
+            {"username": username})
+        course_user_settings = current_user.get("course_settings", {})
+        return self.show_page(courseid, course_user_settings.get(courseid, {}))
+
+    def POST_AUTH(self, courseid):
+        """ POST request """
+        username = self.user_manager.session_username()
+        if not self._is_accessible(courseid, username):
+            return self.template_helper.render("course_unavailable.html")
+        course_user_settings = self._sanitize_content(flask.request.form, courseid)
+        if course_user_settings == {}:
+            return self.show_page(courseid, {"error": ""})
+        self.database.users.update_one({"username": username}, {"$set": {"course_settings." + courseid: course_user_settings}})
+
+        return self.show_page(courseid, course_user_settings)
```

Comment:
Form validation is more implicit than explicit: no success message is shown and no error message indicating what field in invalid neither...